cohort-util/src/main/java/com/ibm/cohort/valueset/ValueSetUtil.java

```diff
@@ -27,6 +27,22 @@
 
 public class ValueSetUtil {
 
+	public static void validateArtifact(ValueSetArtifact valueSetArtifact){
+			if(valueSetArtifact.getUrl() == null){
+				//todo new kind of exception? Invalid Format? something specific
+				throw new IllegalArgumentException("URL must be supplied");
+			}
+			if(valueSetArtifact.getFhirResource().getId() == null || valueSetArtifact.getFhirResource().getId().equals("")){
+				throw new IllegalArgumentException("Identifier must be supplied, ensure that either the OID or the ID field is filled in");
+			}
+			if(valueSetArtifact.getFhirResource().getVersion() == null || valueSetArtifact.getFhirResource().getVersion().equals("")){
+				throw new IllegalArgumentException("Value Set Version must be supplied");
+			}
+			if(valueSetArtifact.getFhirResource().getCompose().getInclude() == null || valueSetArtifact.getFhirResource().getCompose().getInclude().size() == 0){
+				throw new IllegalArgumentException("Value set must include codes but no codes were included.");
+			}
+	}
```

Comment:
Another set of validation checks that might be good to have is to make sure there are no SQL characters in any of the fields. AppScan on Cloud will flag places where we take input parameters and don't do validation on them to avoid SQL Injection attacks. I'm not sure how FHIR handles these fields, or if they would even be getting to a place where any SQL contained in them could be used, but thought I would mention in case it applies.

Reply:
Does that kind of validation not happen in the fhir client itself? I would have assumed as much, since it's the one that actually makes the insertions. Do we do this kind of validation other places that I can see?

Reply:
@tabishop I'm not sure how feasible it is to scan every single field in a complex POJO like this.  Any new fields to either `ValueSetArtifact` or HAPI's `ValueSet` object would render the method obsolete.  Would AppScan be able to tell if we instead attempt to validate against the JSON string that we ultimately send the FHIR server?

Also, do we perform SQL injection tests during our measure/library upload pipeline as well?

Reply:
I think checking the JSON string would probably be ok. I don't have a lot of specifics on how to code the checks that the AppScan tool is flagging. It basically says "you should be doing validation on these input parameters to avoid SQL injection attacks" but it also flags a lot of things that aren't actually problems either. Just wanted to raise it so there is awareness that we may get dinged on it if we don't have an answer at where this is being done.